src/server/serve_dir.rs

```diff
@@ -73,6 +73,11 @@ impl<State> Endpoint<State> for ServeDir {
             // TODO: fix related bug where async-h1 crashes on large files
             let mut res = Response::new(StatusCode::Ok);
             res.set_body(body);
+
+            if let Some(content_type) = mime_guess::from_path(&path).first() {
+                res = res.set_mime(content_type);
```

Comment:
this awkward line is because set_mime takes self. could it it take &mut self?

Reply:
We'll want to replace it with https://docs.rs/http-types/1.2.0/http_types/struct.Response.html#method.set_content_type in a future PR. This requires http-types 2.0 to be released first tho, which is something I'm working on right now. I think for the purpose of this patch this is okay (: